### PR TITLE
test: run rpc-generateblock.py even with wallet disabled

### PR DESCRIPTION
**Pull request by DariusParvin**
Run rpc_generateblock.py test even when the wallet is disabled, as discussed in #20078.

This PR picked up from where #20808 left off, started by @nginocchio. Since then, there have been many improvements to `MiniWallet`, making this PR more straightforward.

L23 makes use of `MiniWallet.rescan_utxos()` to add the pre-mined block utxos (when `self.setup_clean_chain` is not set to `True`), rather than generating new blocks during the test.

### CODE REVIEW

File: test/functional/rpc_generateblock.py [hunk 1]
```diff
@@ -16,72 +17,66 @@ class GenerateBlockTest(BitcoinTestFramework):
...
     def run_test(self):
         node = self.nodes[0]
+        miniwallet = MiniWallet(node)
+        miniwallet.rescan_utxos()
 
         self.log.info('Generate an empty block to address')
-        address = node.getnewaddress()
-        hash = self.generateblock(node, output=address, transactions=[])['hash']
+        address = miniwallet.get_address()
+        hash = node.generateblock(output=address, transactions=[], invalid_call=False)['hash']
```
Comment (MarcoFalke) [inline test/functional/rpc_generateblock.py:27]: This is an invalid call. Can you explain why you are changing this line in the first place?

Comment (DariusParvin) [reply test/functional/rpc_generateblock.py]: ah okay I didn't appreciate the difference I had made. I've changed the calls back to how they were before.

File: test/functional/rpc_generateblock.py [hunk 2]
```diff
@@ -16,15 +17,14 @@ class GenerateBlockTest(BitcoinTestFramework):
...
     def run_test(self):
         node = self.nodes[0]
+        miniwallet = MiniWallet(node)
+        miniwallet.rescan_utxos()
 
         self.log.info('Generate an empty block to address')
-        address = node.getnewaddress()
-        hash = self.generateblock(node, output=address, transactions=[])['hash']
+        address = miniwallet.get_address()
+        hash = self.generateblock(node, address, [])['hash']
```
Comment (MarcoFalke) [inline test/functional/rpc_generateblock.py:27]: why are the named args removed? Again, why is this line changed at all?

Comment (DariusParvin) [reply test/functional/rpc_generateblock.py:27]: The args were not named on the other calls so I removed it here for consistency too. I'll add it back if that it's better to leave it as it.

**Comment by MarcoFalke** [comment]
Thanks, Concept ACK.

**Review by mjdietzx — COMMENTED** [review]
Tested ACK 1e86ff794108d6ef033ceba918b45c5e78c95c62
